src/test/java/org/broadinstitute/hellbender/utils/TranscriptUtilsUnitTest.java

```diff
@@ -0,0 +1,246 @@
+package org.broadinstitute.hellbender.utils;
+
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.ReferenceDataSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.testutils.FuncotatorReferenceTestUtils;
+import org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotationFactory;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class TranscriptUtilsUnitTest extends GATKBaseTest {
+
+    //==================================================================================================================
+    // Private Static Members:
+
+    private static final ReferenceDataSource refDataSourceHg19Ch3;
+    private static final ReferenceDataSource refDataSourceB37;
+
+    private static final List<AutoCloseable> autoCloseableList = new ArrayList<>();
```

Comment:
these tests are reasonable... however they missed the off by 1 bug that was one layer up from this method grabbing the transcrit bases that is not exposed (and possibly not as easy to test because it involves the transcript datasources directly)